Jenkinsfile

```diff
@@ -1,38 +1,229 @@
-// Tell Jenkins how to build projects from this repository
-node {
-    try {
-        properties([
-            [$class: 'BuildDiscarderProperty', strategy: [$class: 'LogRotator', numToKeepStr: '15']]
-        ])
-        
-        stage 'Checkout'
-        checkout scm
-        
-        dir('build') { deleteDir() }
-        dir('.m2/repository/org/eclipse/emf/mwe') { deleteDir() }
-        dir('.m2/repository/org/eclipse/emf/mwe2') { deleteDir() }
-        
-        stage 'Maven Build'
-        def workspace = pwd()
-        def mvnHome = tool 'M3'
-        env.M2_HOME = "${mvnHome}"
-        try {
-            wrap([$class:'Xvnc', useXauthority: true]) {
-                sh "cd maven; ${mvnHome}/bin/mvn -f org.eclipse.emf.mwe2.parent/pom.xml -Declipse.p2.mirrors=false --batch-mode --update-snapshots -fae -Dmaven.repo.local='${workspace}/.m2/repository' -Dtycho.localArtifacts=ignore clean install javadoc:aggregate-jar; ${mvnHome}/bin/mvn -f org.eclipse.emf.mwe2.parent/pom.xml --batch-mode -fae -Dmaven.repo.local='${workspace}/.m2/repository' -Declipse.p2.mirrors=false -DskipTests -Dtycho.localArtifacts=ignore -DaltDeploymentRepository='local::default::file:${workspace}/build/maven-repo' deploy"
+pipeline {
+  agent {
+    kubernetes {
+      label 'migration'
+    }
+  }
+  
+  parameters {
+    choice(name: 'RELEASE_TYPE', choices: ['Integration', 'Beta', 'M1', 'M2', 'M3', 'RC1', 'RC2', 'GA'], description:
+'''Kind of release to build. The chosen value is dependent on the parameter BUILD_TYPE. Use value:
+<ul>
+  <li><tt>Integration</tt>: Continuous Build</li>
+  <li><tt>GA</tt>: Release Build</li>
+  <li>Any other: Stable/Milestone Build</tt></li>  
+</ul>''')
+    booleanParam(
+      name: 'FORCE_PUBLISH',
+      defaultValue: false, 
+      description: 'Force publishing of build artifacts to Eclipse project storage & OSSRH?'
+    )
+  }
+  
+  triggers {
+    cron('H H(0-1) * * *')
```

Comment:
does this aplly to master only?

Reply:
This would apply to all branches. Correct, will be better for master only. I'll change that.